controllers/nugus_controller/nugus_controller.cpp

```diff
@@ -67,24 +67,26 @@ class NUgus : public webots::Robot {
             }
             else if (num_ready > 0) {
                 // Wire format
-                // unit64_t N   message size in bytes
+                // unit32_t N   message size in bytes in network byte order
                 // uint8_t * N  the message
-                uint64_t N;
-                if (recv(tcp_fd, &N, sizeof(N), 0) != sizeof(N)) {
+                uint32_t Nn;
+                if (recv(tcp_fd, &Nn, sizeof(Nn), 0) != sizeof(Nn)) {
                     std::cerr << "Error: Failed to read message size from TCP connection: " << strerror(errno)
                               << std::endl;
                     continue;
                 }
 
-                std::vector<uint8_t> data(N, 0);
-                if (recv(tcp_fd, data.data(), N, 0) != N) {
+				uint32_t Nh = ntohl(Nn);
```

Comment:
I think this deserves a comment of some sort
```suggestion
                // Covert to host endianness, which might be different to network endianness
                uint32_t Nh = ntohl(Nn);
```